openmc/checkvalue.py

```diff
@@ -0,0 +1,337 @@
+            name, value, accepted_values)
+        raise ValueError(msg)
+
+def check_less_than(name, value, maximum, equality=False):
+    """Ensure that an object's value is less than a given value.
+
+    Parameters
+    ----------
+    name : str
+        Description of the value being checked
+    value : object
+        Object to check
+    maximum : object
+        Maximum value to check against
+    equality : bool, optional
+        Whether equality is allowed. Defaults to False.
+
+    """
+
+    if equality:
+        if value > maximum:
+            msg = 'Unable to set "{0}" to "{1}" since it is greater than ' \
+                  '"{2}"'.format(name, value, maximum)
+            raise ValueError(msg)
+    else:
+        if value >= maximum:
+            msg = 'Unable to set "{0}" to "{1}" since it is greater than ' \
+                  'or equal to "{2}"'.format(name, value, maximum)
+            raise ValueError(msg)
+
+def check_greater_than(name, value, minimum, equality=False):
+    """Ensure that an object's value is greater than a given value.
+
+    Parameters
+    ----------
+    name : str
+        Description of the value being checked
+    value : object
+        Object to check
+    minimum : object
+        Minimum value to check against
+    equality : bool, optional
+        Whether equality is allowed. Defaults to False.
+
+    """
+
+    if equality:
+        if value < minimum:
+            msg = 'Unable to set "{0}" to "{1}" since it is less than ' \
+                  '"{2}"'.format(name, value, minimum)
+            raise ValueError(msg)
+    else:
+        if value <= minimum:
+            msg = 'Unable to set "{0}" to "{1}" since it is less than ' \
+                  'or equal to "{2}"'.format(name, value, minimum)
+            raise ValueError(msg)
+
+
+def check_filetype_version(obj, expected_type, expected_version):
+    """Check filetype and version of an HDF5 file.
+
+    Parameters
+    ----------
+    obj : h5py.File
+        HDF5 file to check
+    expected_type : str
+        Expected file type, e.g. 'statepoint'
+    expected_version : int
+        Expected major version number.
+
+    """
+    try:
+        this_filetype = obj.attrs['filetype'].decode()
+        this_version = obj.attrs['version']
+
+        # Check filetype
+        if this_filetype != expected_type:
+            raise IOError('{} is not a {} file.'.format(
+                obj.filename, expected_type))
+
+        # Check version
+        if this_version[0] != expected_version:
+            raise IOError('{} file has a version of {} which is not '
+                          'consistent with the version expected by OpenMC, {}'
+                          .format(this_filetype, 
+                              '.'.join(str(v) for v in this_version),
+                              expected_version))
+    except AttributeError:
+        raise IOError('Could not read {} file. This most likely means the {} '
```

Comment:
There are 2 `{}`s in the formatted string, but only one variable is passed in. Did you mean to add a second variable (possibly, `obj.filename`, but forgot?